src/z3c/insist/enforce.py

```diff
@@ -234,7 +234,7 @@ def registerHandlers(self):
             self.register(handler)
 
     def register(self, handler):
-        self.schedule(handler, path=self.watchedDir, recursive=True)
+        self.schedule(handler, path=self.watchedDir, recursive=True, event_filter=EVENTS_CONSUMED)
```

Comment:
Do we still need checks in dispatch if we have filter here?

Reply:
Don't want to risk it... it basically falls in an endless processing loop when unwanted (esp. EVENT_TYPE_CLOSED_NO_WRITE) come in